source/tnn/device/arm/acc/arm_concat_layer_acc.cc

```diff
@@ -470,12 +478,44 @@ Status ArmConcatLayerAcc::Exec(const std::vector<Blob *> &inputs, const std::vec
     return TNN_OK;
 }
 
+// 修改处：添加了新的函数TransDataType，用于将T_IN类别数据转化为T_OUT类别存储
```

Comment:
Inplace blob data type converting may cause result issues when the blob is referenced by multiple layers. 
Besides, the TransDataType functions was defined in multiple layers, which is not good for maintain, Please add an  optimizer which can insert cast layers to do the converting job.